Components/Queues/PhpAmqpSignalConsumer.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Exception\AMQPRuntimeException;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Handler\AmqpQueueHandler;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Handler\PhpAmqpHandler;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\ConsumerInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\IsStopableConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
+use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpSignalConsumer
+ * @package Smartbox\Integration\FrameworkBundle\Components\Queues
+ */
+class PhpAmqpSignalConsumer extends Service implements ConsumerInterface, LoggerAwareInterface
+{
+    use IsStopableConsumer;
+    use UsesSmartesbHelper;
+    use UsesSerializer;
+    use LoggerAwareTrait;
+    use UsesExceptionHandlerTrait;
```

Comment:
Are all these traits needed?

Reply:
The traits was validated. The unused ones was removed.